fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt

```diff
@@ -0,0 +1,248 @@
+    private val interactor: DefaultOnboardingInteractor
+        get() = _interactor!!
+
+    private var onboardingView: OnboardingView? = null
+    private var homeMenuView: HomeMenuView? = null
+    private var tabCounterView: TabCounterView? = null
+    private var toolbarView: ToolbarView? = null
+
+    private lateinit var onboardingStore: OnboardingStore
+    private lateinit var onboardingAccountObserver: OnboardingAccountObserver
+
+    private val searchSelectorBinding = ViewBoundFeatureWrapper<SearchSelectorBinding>()
+    private val searchSelectorMenuBinding = ViewBoundFeatureWrapper<SearchSelectorMenuBinding>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = FragmentHomeBinding.inflate(inflater, container, false)
+
+        val activity = activity as HomeActivity
+
+        onboardingAccountObserver = OnboardingAccountObserver(
+            context = requireContext(),
+            dispatchChanges = ::dispatchOnboardingStateChanges,
+        )
+
+        onboardingStore = StoreProvider.get(this) {
+            OnboardingStore(
+                initialState = OnboardingFragmentState(
+                    onboardingState = onboardingAccountObserver.getOnboardingState(),
+                ),
+            )
+        }
+
+        _interactor = DefaultOnboardingInteractor(
+            controller = DefaultOnboardingController(
+                activity = activity,
+                navController = findNavController(),
+                hideOnboarding = ::hideOnboarding,
+            ),
+            privateBrowsingController = DefaultPrivateBrowsingController(
+                activity = activity,
+                appStore = requireComponents.appStore,
+                navController = findNavController(),
+            ),
+            searchSelectorController = DefaultSearchSelectorController(
+                activity = activity,
+                navController = findNavController(),
+            ),
+            toolbarController = DefaultToolbarController(
+                activity = activity,
+                store = requireComponents.core.store,
+                navController = findNavController(),
+            ),
+        )
+
+        toolbarView = ToolbarView(
+            binding = binding,
+            context = requireContext(),
+            interactor = interactor,
+        )
+
+        onboardingView = OnboardingView(
+            containerView = binding.sessionControlRecyclerView,
+            interactor = interactor,
+        )
+
+        activity.themeManager.applyStatusBarTheme(activity)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        consumeFrom(onboardingStore) { state ->
+            onboardingView?.update(state.onboardingState, requireComponents.fenixOnboarding.config)
```

Comment:
👍 

Curious, are Nimbus configs only accessible via `requireComponents`?

Reply:
This is more about Nimbus configs being inside of  [FenixOnboarding#35](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/onboarding/FenixOnboarding.kt#35), which has been made into a singleton that is accessible only through the (Components#170)[https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt#170].